client/src/components/annotators/ImageAnnotator.vue

```diff
@@ -181,40 +192,49 @@ export default Vue.extend({
      * @param {int} max upper bound frame number for caching
      */
     cacheNewRange(min, max) {
+      const frameImages = [];
       for (let i = this.frame; i <= max; i += 1) {
-        this.cacheFrame(i);
+        frameImages[i] = this.cacheFrame(i);
         const minusFrame = this.frame - (i - this.frame);
         if (minusFrame >= min) {
           this.cacheFrame(minusFrame);
         }
       }
+      return frameImages;
     },
     /**
      * Adds a single frame to the pendingImgs array for loading and assigns it to the master
      * imgs list. Once the image is loaded it is removed from the pendingImgs
      * @param {int} i the image to cache if it isn't already assigned
+     * @returns {Array<Promise<Array<imgage>>>} loaded images
      */
     cacheFrame(i) {
+      let p = Promise.resolve(this.imgs[i]);
       if (!this.imgs[i]) {
         const img = new Image();
         img.crossOrigin = 'Anonymous';
         this.imgs[i] = img;
         const imageAndFrame = [img, i];
         this.pendingImgs.add(imageAndFrame);
-        img.onload = () => {
-          this.pendingImgs.delete(imageAndFrame);
-          img.onload = null;
-          img.cached = true;
-          // If we are trying to play and waiting for loaded frames we check the cache again
-          if (this.playing && this.loadingVideo) {
-            if (this.checkCached(this.playCache)) {
-              this.loadingVideo = false;
-              this.syncWithVideo();
+        p = new Promise((resolve) => {
+          img.onload = (evt) => {
```

Comment:
```suggestion
          img.onload = () => {
```
unsued variable, Assuming it was there for testing before?